### DISCUSSION TITLE
Optimal settings for iBeacon Tag

### DISCUSSION THREAD

**Discussion by etcg** [mKeRix/room-assistant#334] [Q&A]
Hi,

I got an Mineaw E8 Tag [(Link)](https://www.minew.com/products/e8-tag-beacon.html), which I try to set-up as an iBeacon with the corresponding App. Can you guys help me to figure out the right settings in the app and for the room-assistant config? Range should be about ~5-7 meters and maybe tweak it to save as much battery as possible (I disabled all other protocols except iBeacon). The tag has an accelerometer sensor, which I disabled, because I don't need it. Thanks!

![image0 (2)](https://user-images.githubusercontent.com/50837349/97985859-232a1000-1dd9-11eb-8b78-defdad417527.jpeg)

**Comment by mKeRix** [comment]
I think your settings look quite nice already! The biggest factor for battery saving in my eyes is the advertising interval, so if you don't need super fast reaction times you could also increase that a bit. Other than that I would measure the RSSI @1m by putting it exactly 1m (free line of sight) away from one of your room-assistant instances and then noting the value that room-assistant reports. You can do so by either watching the log while room-assistant starts or by going to http://<room-assistant-ip>:6415/entities and looking at the respective `rssi` in `measuredValues `. Maybe refresh that endpoint a few time to get a good average.

On the room-assistant side of things you shouldn't need much more than adding your tag id to the `whitelist` of the BLE integration. If you increasing the advertising interval you may also need to increase the `timeout` in room-assistant, I think this setting should at least be double of the advertising interval.

EDIT: And the always advertise switch may need to be toggled on in your app.

**Reply by etcg** [reply]
Thank you for your answer! Yes, the advertising interval is really too low, I changed that to 3 seconds now. Even 5-10 would be enough for me. But I don't get what I should do with the measured RSSI value. Should I set it in the App or in the RA config? You just said noting it. I only have one room-assistant instance, because for now I just use it for general presence detection. I think the always advertise setting is for the trigger (motion) section, because I can't activate it without the motion trigger.

**Reply by mKeRix** [reply]
Since your beacon supports it and you have the nice interface in the app I would just set the value there once you figured it out. room-assistant will pick it up automatically from the beacon. The override you can set in the room-assistant config is more meant for devices where you cannot configure this value.

**Reply by etcg** [reply]
Okay I set it all up and its running and working. Thanks for your help and great project by the way. I used my iPhone with bluetooth classic integration before, but the battery drain was really insane. If I spent a whole day at home I needed to recharge early in the evening.   Before I used RA and now again with the tag it's back to the normal 1.5 day use. Just for reference I use a pi zero w, the installation process was slow as hell (which obviously isn't the fault of RA), but after that its working without problems. 

As advice for other pi users: I installed [log2ram](https://github.com/azlux/log2ram), so the SD card doesn't wear out too fast.

**Reply by mKeRix** [reply]
+1 for the log2ram suggestion, I discovered that project a couple months back and now also use it on all my Pis. :)

I actually hope that the installation will become a bit quicker with some changes I recently developed for the thermopile sensors, as those get rid of some native dependencies that compile for a long time.
